optax/_src/loss.py

```diff
@@ -503,3 +503,19 @@ def kl_divergence_with_log_targets(log_predictions: chex.Array,
   chex.assert_type([log_predictions, log_targets], float)
   loss = jnp.exp(log_targets) * (log_targets - log_predictions)
   return jnp.sum(loss, axis=-1)
+
+def hinge_loss(predictions: chex.Array, targets: chex.Array) -> chex.Array:
+  """
+  Computes the hinge loss for Binary Classification.
+
+  Args:
+    predictions: Predicted values with shape [..., dim].
+    targets: Target values with shape [..., dim].
+      Expected to be in range [-1, 1].
+
+  Returns:
+    Average Hinge Loss
+  """
+  assert predictions.shape[0] == targets.shape[0], "Predictions and Target lengths must be consistent."
```

Comment:
I don't think we need this check in the binary case so I'd remove this line. We'd only need it if we want to check that the number of classes matches in the multi-class setting. In this case, we can just let jax numpy raise an error if it can't broadcast the arrays to compatible shapes below.